FH-3807 Update template for Template Service Broker

aidenkeating opened the pull request.

Commit:
FH-3807 Update template for Template Service Broker

This commit does a few things:
* Cleans up some unneeded data to make the file more readable.
* Removes previous 'expose' annotations.
* Adds 'expose' annotation for public URL for the template binding.
* Adds more annotations for template metadata.

aidenkeating commented:
@david-martin Would you mind taking a look? 

I'm not sure if those `status` attributes meant something that I might have missed or whether those template expose annotations in the secret were meant to be there. This leaves a template binding with one value, the public URL which sync is accessible at.

The metadata might need to be updated also, I just added some stuff.

david-martin commented on fh-sync-server-DEVELOPMENT.yaml:
```diff
@@ -3,6 +3,17 @@ kind: Template
+    iconClass: icon-nodejs
+    tags: instant-app,sync
+    template.openshift.io/long-description: Data Synchronisation by FeedHenry.
+    template.openshift.io/provider-display-name: Red Hat, Inc.
```
as this is upstream, does it make sense to have Red Hat? I would think FeedHenry would be OK here.

david-martin commented on fh-sync-server-DEVELOPMENT.yaml:
```diff
@@ -343,27 +331,16 @@ objects:
 - apiVersion: v1
+  kind: Secret
   data:
     database-admin-password: dUxrSmZQbXNqOFh4dmNpcg==
```
probably a follow up issue, but we should generate these values rather than hardcoded

david-martin reviewed (approved):
This template was largely an 'oc export' of the resources after getting them running in an openshift cluster. Removing the unnecessary fields and exposed values makes sense

Commit:
FH-3807 Generate MongoDB credentials if not provided as parameters

Changes the hardcoded values for MongoDB to generated ones. These
can also be specified when deploying the template from the Catalog
if the user wishes to override them.
